src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.TrailingHeaders.cs

```diff
@@ -0,0 +1,252 @@
+                        }
+
+                        Assert.Contains("amazingtrailer", response.TrailingHeaders.GetValues("MyCoolTrailerHeader"));
+                        Assert.Contains("World", response.TrailingHeaders.GetValues("Hello"));
+
+                        string data = await response.Content.ReadAsStringAsync();
+                        Assert.Contains("data", data);
+                        // Trailers should not be part of the content data.
+                        Assert.DoesNotContain("MyCoolTrailerHeader", data);
+                        Assert.DoesNotContain("amazingtrailer", data);
+                        Assert.DoesNotContain("Hello", data);
+                        Assert.DoesNotContain("World", data);
+                    }
+                }
+            });
+        }
+
+        [Fact]
+        public async Task GetAsyncResponseHeadersReadOption_TrailingHeaders_Available()
+        {
+            await LoopbackServer.CreateServerAsync(async (server, url) =>
+            {
+                using (HttpClientHandler handler = CreateHttpClientHandler())
+                using (var client = new HttpClient(handler))
+                {
+                    Task<HttpResponseMessage> getResponseTask = client.GetAsync(url, HttpCompletionOption.ResponseHeadersRead);
+                    await TestHelper.WhenAllCompletedOrAnyFailed(
+                        getResponseTask,
+                        server.AcceptConnectionSendCustomResponseAndCloseAsync(
+                            "HTTP/1.1 200 OK\r\n" +
+                            "Connection: close\r\n" +
+                            "Transfer-Encoding: chunked\r\n" +
+                            "Trailer: MyCoolTrailerHeader\r\n" +
+                            "\r\n" +
+                            "4\r\n" +
+                            "data\r\n" +
+                            "0\r\n" +
+                            "MyCoolTrailerHeader: amazingtrailer\r\n" +
+                            "Hello: World\r\n" +
+                            "\r\n"));
+
+                    using (HttpResponseMessage response = await getResponseTask)
+                    {
+                        Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+                        Assert.Contains("chunked", response.Headers.GetValues("Transfer-Encoding"));
+                        Assert.Contains("MyCoolTrailerHeader", response.Headers.GetValues("Trailer"));
+
+                        // Pending read on the response content.
+                        var trailingHeaders = response.TrailingHeaders;
+                        Assert.Empty(trailingHeaders);
+
+                        Stream stream = await response.Content.ReadAsStreamAsync();
+                        Byte[] data = new Byte[100];
+                        // Read some data, preferably whole body.
+                        await stream.ReadAsync(data, 0, 4);
```

Comment:
I can't remember if I commented on this previously, but there's no guarantee this is actually going to read all 4 bytes.

Reply:
yes, you did. This test will work even if we read less. Preferably, this test will consume whole body but if not the next section will as it reads in loop to the end.

Reply:
>  This test will work even if we read less

I don't see how.  Two lines down the code does:
```C#
Assert.Contains("data", System.Text.Encoding.Default.GetString(data));
```
If this line reads less than the requested 4 bytes, that assert will fail.

Reply:
That is because I did not push one of my local changes ;(
I was originally thinking about loop around ReadAsync but decided to make body check conditional as that is not crux of the test.